src/core/algorithms/algo_factory.h

```diff
@@ -77,6 +78,36 @@ void LoadAlgorithm(Algorithm& algorithm, OptionMap&& options) {
                     details::ExtractOptionValue<char>(options, kSeparator),
                     details::ExtractOptionValue<bool>(options, kHasHeader));
             return boost::any{parser};
+        } else if (option_name == kTableSet
+                   && options.find(std::string{kTableSet}) == options.end()) {
```

Comment:
There is also an interesting question: should we introduce a new option `--csv_paths` for multiple files or reuse the existing `--csv_path` field.
It looks like we'll have to introduce a new option. Otherwise, it will be necessary to change the existing code (for example, in tests the `--csv_path` parameter is often set and a single path is passed, but it will be necessary to pass a vector of paths)
But on the other hand, it may not be obvious to the user which option from the console to use when he wants to run an algorithm on one dataset that expects several datasets.
It may be necessary to be able to allow the `TableSet` to take into account the values of both options.